src/utils/geometry.rs

```diff
@@ -833,6 +833,19 @@ impl<N: Coordinate, Kind> Rectangle<N, Kind> {
         )
     }
 
+    /// Clamp rectangle to min and max corners resulting in the overlapping area of two rectangles
+    #[inline]
+    pub fn overlapping_area(self, other: impl Into<Rectangle<N, Kind>>) -> Self {
```

Comment:
Nitpick, but I tend to think naming this "intersection" would be more succinct and slightly clearer.